pkg/common/oras/authprovider/alibabacloud/alibabacloudacrbasic.go

```diff
@@ -0,0 +1,233 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package alibabacloud
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	cr20181201 "github.com/alibabacloud-go/cr-20181201/v2/client"
+	openapi "github.com/alibabacloud-go/darabonba-openapi/v2/client"
+	util "github.com/alibabacloud-go/tea-utils/v2/service"
+	"github.com/alibabacloud-go/tea/tea"
+	"github.com/aliyun/credentials-go/credentials"
+	re "github.com/ratify-project/ratify/errors"
+
+	"github.com/pkg/errors"
+	provider "github.com/ratify-project/ratify/pkg/common/oras/authprovider"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	EnvRoleArn              = "ALIBABA_CLOUD_ROLE_ARN"
+	EnvOidcProviderArn      = "ALIBABA_CLOUD_OIDC_PROVIDER_ARN"
+	EnvOidcTokenFile        = "ALIBABA_CLOUD_OIDC_TOKEN_FILE"
+	AlibabaCloudACREndpoint = "cr.%s.aliyuncs.com"
+)
+
+type AlibabaCloudAcrBasicProviderFactory struct{} //nolint:revive // ignore linter to have unique type name
+
+type acrInstancesConfig struct {
+	InstanceName string `json:"instanceName"`
+	InstanceID   string `json:"instanceId"`
+}
+
```

Comment:
will Alibaba auth provider consider adding metric reporting in future improvement PRs? https://github.com/ratify-project/ratify/blob/8647de290fa0b150ae9082f68ff03dd91f2bff7a/pkg/common/oras/authprovider/azure/azureworkloadidentity.go#L55

Reply:
Sure, maybe you can leave an issue for future tracking.